staging/src/k8s.io/api/networking/v1/well_known_annotations.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+const (
+	// AnnotationIsDefaultIngressClass can be used to indicate that an
+	// IngressClass should be considered default. When a single IngressClass
+	// resource has this annotation set to true, new Ingress resources without a
+	// class specified will be assigned this default class.
+	AnnotationIsDefaultIngressClass = "ingressclass.kubernetes.io/is-default-class"
+
+	// AnnotationIngressClass indicates the class of an Ingress to be used when
+	// determining which controller should implement the Ingress. Use of this
+	// annotation is deprecated. The Ingress class field should be used instead
+	// of this annotation.
+	// +deprecated
+	AnnotationIngressClass = "kubernetes.io/ingress.class"
```

Comment:
I'm not sure what the best practice is for a deprecated annotation like this. I think we still want to keep the current behavior that prevents an Ingress from being created if it specifies both this annotation and the class field, but it does feel a bit weird to "graduate" a deprecated annotation to v1. Maybe it needs to be part of v1 since we're still using the value for validation?

Reply:
validation is not using the exported API constant, it has its own private value

I would probably use the exported v1beta1 constant, not redefine it in v1.